base/algorithms.jl

```diff
@@ -0,0 +1,119 @@
+module Algorithms
+
+export replace!, replace
+
+import Base: replace
+
+
+"""
+    replace!(pred, [f::Function], A, [new], [count])
+
+Replace all occurrences `x` in collection `A` for which `pred(x)` is true
+by `new` or `f(x)` (exactly one among `f` and `new` must be specified).
+If `count` is specified, then replace at most `count` occurrences.
+This is the in-place version of [`replace`](@ref).
+
+# Examples
+```jldoctest
+julia> a = [1, 2, 3, 1];
+
+julia> replace!(isodd, a, 0, 2); a
```

Comment:
It was to clarify that `a` is modified in-place, but this may not be necessary. As to return `count`, I see that it could be useful, but I'm uncomfortable with the inconsistency of having the non-mutating `replace` returning the collection instead.